File: src/java/SpyNumber.java
package java;

public class SpyNumber {

    public static boolean isSpyNumber(long num) {
        int sumOfDigit = 0, productOfDigits = 1;

        int dig;
        while(num > 0) {
            dig = (int) num % 10;
            sumOfDigit += dig;
            productOfDigits *= dig;
            num /= 10;
        }

        return sumOfDigit == productOfDigits;
    }

}

File: src/javascript/spyNumber.js
const isSpyNumber = (num) => {
  let sumOfDigits = 0;
  let productOfDigits = 1;

  while (num > 0) {
    const dig = num % 10;
    sumOfDigits += dig;
    productOfDigits *= dig;
    num = Math.floor(num / 10);
  }

  return sumOfDigits === productOfDigits;
}

export default isSpyNumber;
